### PR TITLE
Fix: Pep8 fixes for /agents

### PR DESCRIPTION
**Pull request by hTrap**
## Description
Pep8 fixes for /agents submitted as a seperate PR as discussed in #623 

## Reviewers

@delta24 @DePierre 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Other

### Checklist:

- [x] My code follows the code style (modified PEP8) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

Signed-off-by: Parth Oberoi <htrapdev@gmail.com>

### CODE REVIEW

**Comment by DePierre** [comment]
Hi @hTrap, it is fairly easier to review the changes, thank you for creating separate PR! :)

In your changes, there are still many pep8 violations. For instance the naming convention (CamelCase). I understand that changing the name of the functions is too tedious to be done here, but at least we could update the naming convention of the internal variables.

I have pin-pointed one in the PR. I also commented a couple other stuff. They apply to the other PR (#623 for instance).

File: agents/imap/general.py [hunk 1]
```diff
@@ -3,78 +3,85 @@
...
+import os
+import sys
 from collections import defaultdict
 
+
 class Plugin:
-	def Get(self, ModuleName, ModuleFile, ModulePath):# Python fiddling to load a module from a file, there is probably a better way...
-		f, Filename, desc = imp.find_module(ModuleFile.split('.')[0], [ModulePath]) #ModulePath = os.path.abspath(ModuleFile)
-		return imp.load_module(ModuleName, f, Filename, desc)
+    # Python fiddling to load a module from a file, there is probably a better way...
```
Comment (DePierre) [inline agents/imap/general.py]: This should be a docstring and not a comment

File: agents/imap/general.py [hunk 2]
```diff
@@ -3,78 +3,85 @@
...
+    def Get(self, ModuleName, ModuleFile, ModulePath):
+        f, Filename, desc = imp.find_module(ModuleFile.split('.')[0], [ModulePath])
```
Comment (DePierre) [inline agents/imap/general.py]: Camel Case

File: agents/imap/general.py [hunk 3]
```diff
@@ -3,78 +3,85 @@
...
+        return imp.load_module(ModuleName, f, Filename, desc)
+
+    def Run(self, Name, Path, Params):
+        return self.Get("", Name, "%s/" % Path).Run(Params)
```
Comment (DePierre) [inline agents/imap/general.py]: Ditto

File: agents/imap/general.py [hunk 4]
```diff
@@ -3,78 +3,85 @@
...
+        self.Config = defaultdict(list)
+        self.Load(FileName)
+
+    def Load(self, FileName):
+        for Line in File().GetAsList(FileName):
+            try:
+                Name = Line.split(":")[0]
+                Value = Line.replace("%s:" % Name, "").strip()
+                self.Config[Name] = Value
+            except:
```
Comment (DePierre) [inline agents/imap/general.py]: Catching all exceptions is really a bad habit. Could you update the `except` to catch only the required exceptions?

File: agents/imap/general.py [hunk 5]
```diff
@@ -1,80 +1,88 @@
...
+
 class Plugin:
-	def Get(self, ModuleName, ModuleFile, ModulePath):# Python fiddling to load a module from a file, there is probably a better way...
-		f, Filename, desc = imp.find_module(ModuleFile.split('.')[0], [ModulePath]) #ModulePath = os.path.abspath(ModuleFile)
-		return imp.load_module(ModuleName, f, Filename, desc)
+    '''
+    Python fiddling to load a module from a file, there is probably a better way...
+    '''
+    def Get(self, ModuleName, ModuleFile, ModulePath):
+        f, FileName, desc = imp.find_module(ModuleFile.split('.')[0], [ModulePath])
```
Comment (DePierre) [inline agents/imap/general.py]: `FileName` is not appropriate here.

File: agents/imap/general.py [hunk 6]
```diff
@@ -1,80 +1,88 @@
...
-		if not os.path.isfile(self.FileName):
-			self.Init()
-		else:
-			self.Set(File().GetAsList(self.FileName)[0].strip())
-	
-	def Save(self):
-		#print "Saving Storage " + str(self.Content) + ".."
-		File().Save(self.FileName, self.Content)
+    def __init__(self, FileName):
+        self.FileName = FileName
```
Comment (DePierre) [inline agents/imap/general.py]: Ditto

File: agents/imap/general.py [hunk 7]
```diff
@@ -1,80 +1,88 @@
...
-	def Save(self, FileName, Data):
-		Data = str(Data)
-		try:
-			print "FileName=" + str(FileName) + ", Data=" + str(Data)
-			print "Saving to File '" + str(FileName) + "' " + str(Data) + ".."
-			File = open(FileName, 'w')
-			File.write(Data)
-		except IOError, error:
-			print "Cannot write to: '" + FileName +"' ("+str(sys.exc_info())+")"
+    def GetAsList(self, FileName):
```
Comment (DePierre) [inline agents/imap/general.py]: Ditto

File: agents/imap/general.py [hunk 8]
```diff
@@ -1,80 +1,88 @@
...
+        self.Config = defaultdict(list)
+        self.Load(FileName)
+
+    def Load(self, FileName):
+        for line in File().GetAsList(FileName):
+            try:
+                name = line.split(":")[0]
+                value = line.replace("%s:" % name, "").strip()
+                self.Config[name] = value
+            except ValueError:
```
Comment (DePierre) [inline agents/imap/general.py]: Could you help me understand why we are expecting `ValueError` to occur?

File: agents/imap/imapd.py [hunk 9]
```diff
@@ -3,41 +3,44 @@
...
+            connection.close()
+            connection.logout()
+            storage.Save()
+            print "Sleeping %s seconds..(Control+C to stop agent)" % config.Get('WAIT_SECS')
+            time.sleep(int(config.Get('WAIT_SECS')))
+    except KeyboardInterrupt:
+        print "Stopping daemon.."
+        storage.Save()
+
+config = Config('config.cfg')
```
Comment (DePierre) [inline agents/imap/imapd.py]: Are we sure these two global variables are not used somewhere else in owtf?

File: agents/imap/payloads/link_clicker.py [hunk 10]
```diff
@@ -3,17 +3,20 @@
...
-		VisitURLCmd = "curl " + URL
-		subprocess.Popen(VisitURLCmd, shell=True) # Visit URL
-		# Log visit:
-		subprocess.Popen('echo "$(date)" - ' + VisitURLCmd + LogStr, shell=True)
+
+def Run(params):
+    LogStr = " >> %s 2>> %s" % (params['Log'], params['ErrorLog'])
+    for URL in re.findall(URL_REGEX, str(params['Message'])):
+        print "Found URL=%s" % URL
+        VisitURLCmd = "curl %s" % URL
```
Comment (DePierre) [inline agents/imap/payloads/link_clicker.py]: Camel Case

File: agents/imap/payloads/link_clicker.py [hunk 11]
```diff
@@ -3,17 +3,20 @@
...
-	LogStr = " >> " + Params['Log'] + " 2>> " + Params['ErrorLog']
-	for URL in re.findall(URL_REGEX, str(Params['Message'])):
-		print "Found URL=" + URL
-		VisitURLCmd = "curl " + URL
-		subprocess.Popen(VisitURLCmd, shell=True) # Visit URL
-		# Log visit:
-		subprocess.Popen('echo "$(date)" - ' + VisitURLCmd + LogStr, shell=True)
+
+def Run(params):
+    LogStr = " >> %s 2>> %s" % (params['Log'], params['ErrorLog'])
```
Comment (DePierre) [inline agents/imap/payloads/link_clicker.py]: Camel Case

File: agents/imap/general.py (hunk 8 above)
Comment (hTrap) [inline agents/imap/general.py]: This would occur in a split if there is no ':' in a line

Comment (DePierre) [inline agents/imap/general.py]: @hTrap Are you sure about that?
```pycli
>>> line = ''
>>> line.split(':')[0]
''
```

File: agents/imap/general.py (hunk 5 above)
Comment (hTrap) [inline agents/imap/general.py]: UpperCamelCase or lowerCamelCase ?

**Comment by DePierre** [comment]
Close/re-open to trigger Travis.

File: agents/imap/general.py (hunk 5 above)
Comment (DePierre) [inline agents/imap/general.py]: None of the above. Use ``lower_case_with_underscores`` instead please :)

**Comment by delta24** [comment]
@hTrap Can you update the PR as per the comments?

**Pull request closed without merging**